Content.Server/Atmos/EntitySystems/HeatExchangerSystem.cs

```diff
@@ -77,29 +77,54 @@ private void OnAtmosUpdate(EntityUid uid, HeatExchangerComponent comp, AtmosDevi
         else
             xfer = outlet.Air.Remove(-dN);
 
+        float CXfer = _atmosphereSystem.GetHeatCapacity(xfer);
+        if (CXfer < Atmospherics.MinimumHeatCapacity)
+            return;
+
         var radTemp = Atmospherics.TCMB;
 
-        // Convection
         var environment = _atmosphereSystem.GetContainingMixture(uid, true, true);
-        if (environment != null && environment.TotalMoles != 0)
+        bool hasEnv = false;
+        float CEnv = 0f;
+        if (environment != null)
         {
-            radTemp = environment.Temperature;
-
-            // Positive dT is from pipe to surroundings
-            var dT = xfer.Temperature - environment.Temperature;
-            var dE = comp.K * dT * dt;
-            var envLim = Math.Abs(_atmosphereSystem.GetHeatCapacity(environment) * dT * dt);
-            var xferLim = Math.Abs(_atmosphereSystem.GetHeatCapacity(xfer) * dT * dt);
-            var dEactual = Math.Sign(dE) * Math.Min(Math.Abs(dE), Math.Min(envLim, xferLim));
-            _atmosphereSystem.AddHeat(xfer, -dEactual);
-            _atmosphereSystem.AddHeat(environment, dEactual);
+            CEnv = _atmosphereSystem.GetHeatCapacity(environment);
+            hasEnv = CEnv >= Atmospherics.MinimumHeatCapacity && environment.TotalMoles > 0f;
```

Comment:
If `TotalMoles` == 0, then can `CEnv` be > `Atmospherics.MinimumHeatCapacity`?

Reply:
yes, it can be, i tested